src/CollateralizedVault.sol

```diff
@@ -0,0 +1,237 @@
+        userRatio = WDiv.wdiv(deposits[msg.sender][WETH], _totalDebt(user));
+    }
+
+    function ratio(address user) view external returns(uint256 userRatio) {
+        userRatio = _ratio(user);
+    }
+
+    /// @notice allows any user to liquidate under collateralized users
+    function liquidate(address user) external returns(uint256){
+        require(!_isCollateralized(user), "specified user can't be liquidated");
+        uint256 lqdtdWETH = deposits[user][WETH];
+        uint256 debtDAI = debts[user][DAI];
+        uint256 debtUSDC = debts[user][USDC];
+        deposits[user][WETH] = 0;
+        debts[user][DAI] = 0;
+        debts[user][USDC] = 0;
+        IERC20(DAI).safeTransferFrom(msg.sender, address(this), debtDAI);
+        IERC20(USDC).safeTransferFrom(msg.sender, address(this), debtUSDC);
+        IERC20(WETH).safeTransfer(msg.sender, lqdtdWETH);
+        return lqdtdWETH;
+    }
+
+    ///@notice function to determine if a user is currently sufficiently collateralized
+    ///@dev max ltv ration is 66%
+    function _isCollateralized(address user) internal view returns(bool collateralized) {
+        uint256 debt = _totalDebt(user);
+        uint256 deposit_ = deposits[user][WETH];
+
+        uint256 vaultRatio = WDiv.wdiv(debt, deposit_);
+
+        if(vaultRatio <= 66e16){
+            collateralized = true;
+        } else {
+            collateralized = false;
+        }
+        
+    }
+
+    /// @return price the price of 1 DAI in WETH
+    function _daiPrice() internal view returns(uint256) {
+        (   ,
+        int256 price,
+            ,
+            ,
+        ) = priceFeeds[DAI].latestRoundData();
+        return uint256(price);
+    }
+
+    /// @return price the price of 1 USDC in WETH
+    function _usdcPrice() internal view returns(uint256) {
+        (   /*uint80 roundID*/,
+        int256 price,
+            /*uint startedAt*/,
+            /*uint timeStam*/,
+            /*uint80 answeredInRound*/
+        ) = priceFeeds[USDC].latestRoundData();
+        return uint256(price);
+    }
+
+    function _availableDAI(address user) internal view returns(uint256) {
+        uint256 freeCollateral = deposits[user][WETH] - _totalDebt(user);
+        
+        return WDiv.wdiv(freeCollateral, _daiPrice());
+    }
+
+    function _availableUSDC(address user) internal view returns(uint256) {
+        uint256 freeCollateral = deposits[user][WETH] - _totalDebt(user);
+        return WDiv.wdiv(freeCollateral, _usdcPrice());
+    }
+
+
+    /// @dev because USDC is six decimals, 12 decimals are added to the USDC/WETH amount befoe adding the DAI/WETH amount
+    function _totalDebt(address user) internal view returns(uint256) {
+       return _dai2WETH(debts[user][DAI]) + (_usdc2WETH(debts[user][USDC])*10**12);
+    }
+
+    /// @dev provides how much an amount of DAI is worth in WETH
+    function _dai2WETH(uint256 amount) internal view returns(uint256) {
+        return WMul.wmul(amount, _daiPrice());
```

Comment:
_daiPrice() returns how much 1 DAI is worth in ETH. So assuming perfect peg, if ETH is trading at $2000, and you put in 500 DAI for the amount, it would return .25 ETH